cpp/oneapi/dal/algo/pca/backend/gpu/finalize_train_kernel_cov_impl_dpc.cpp

```diff
@@ -98,19 +98,25 @@ result_t finalize_train_kernel_cov_impl<Float>::operator()(const descriptor_t& d
             pr::correlation_from_covariance(q, rows_count_global, cov, corr, bias, { cov_event });
         data_to_compute = corr;
     }
-
-    auto [eigvecs, eigvals] = compute_eigenvectors_on_host(q,
-                                                           std::move(data_to_compute),
-                                                           component_count,
-                                                           { corr_event, vars_event, cov_event });
+    cov_event.wait_and_throw();
+    vars_event.wait_and_throw();
+    corr_event.wait_and_throw();
+    auto [eigvals, syevd_event] =
+        syevd_computation(q, data_to_compute, { cov_event, corr_event, vars_event });
+    syevd_event.wait_and_throw();
```

Comment:
I guess event dependencies must be reviewed here to avoid unnecessary wait_and_throw